h2o-core/src/main/java/hex/Model.java

```diff
@@ -1066,14 +1061,32 @@ public Frame score(Frame fr, String destination_key, Job j) throws IllegalArgume
     return score(fr, destination_key, j, true);
   }
 
+  public void addWarning(String s){
+    _warnings = Arrays.copyOf(_warnings,_warnings.length+1);
+    _warnings[_warnings.length-1] = s;
+  }
+
+  public void addWarningP(String s){
+    _warningsP = Arrays.copyOf(_warningsP,_warningsP.length+1);
+    _warningsP[_warningsP.length-1] = s;
+  }
+
   public Frame score(Frame fr, String destination_key, Job j, boolean computeMetrics) throws IllegalArgumentException {
     Frame adaptFr = new Frame(fr);
     computeMetrics = computeMetrics && (!isSupervised() || (adaptFr.vec(_output.responseName()) != null && !adaptFr.vec(_output.responseName()).isBad()));
     String[] msg = adaptTestForTrain(adaptFr,true, computeMetrics);   // Adapt
+    // clean up the previous score warning messages
+    _warningsP = new String[0];
     if (msg.length > 0) {
-      for (String s : msg)
-        Log.warn(s);
+      for (String s : msg) {
+        if (!(this.isSupervised() && s.contains(this._parms._response_column))) {
```

Comment:
I think a better way might be not to generated a warning about the response column in the adaptTestForTrain function, so instead of generating the warning and the filtering it out - just don't generate the message at all.

This would need a flag in the adaptTestForTrain function or some other indicator.